libbeat/api/routes.go

```diff
@@ -38,6 +40,10 @@ func NewWithDefaultRoutes(log *logp.Logger, config *common.Config, ns lookupFunc
 	mux.HandleFunc("/state", makeAPIHandler(ns("state")))
 	mux.HandleFunc("/stats", makeAPIHandler(ns("stats")))
 	mux.HandleFunc("/dataset", makeAPIHandler(ns("dataset")))
```

Comment:
I know this PR is already merged but I just noticed this while reviewing the backport PR. Should these three API routes be added via `AddHandlerFunc` too? That would ensure that someone doesn't try to override these routes when registering a custom one.

Reply:
Thanks for the comments.

As mentioned in this [reply](https://github.com/elastic/beats/pull/22605#issuecomment-729666567), the usage of `AddHandlerFunc` would be registering other routes in a module's "init" phase, that makes customized routes to be shown up before "default" routes, and the reason "default" routes can not be done in "init" phase is the handler functions relies on `ns lookupFunc` which is passed only in `NewWithDefaultRoutes`.

I think what can be done is 
* call `AddHandlerFunc` for default routes as well so at least errors can be logged when there are conflicts.
* or move default routes registration to init phase and hardcode the `ns` to be `monitoring.GetNamespace` and change the interface of `NewWithDefaultRoutes `.

Any thoughts?